libs/azure-iot-middleware-freertos/source/azure_iot_adu_client.c

```diff
@@ -311,18 +309,16 @@ AzureIoTResult_t AzureIoTADUClient_SendResponse( AzureIoTADUClient_t * pxAzureIo
         return AzureIoT_TranslateCoreError( xAzResult );
     }
 
-    /* TODO: consume xRequestDecision in az_iot_adu_client_get_service_properties_response */
-    ( void ) xRequestDecision;
+    /* Code 406 is "Not Acceptable" */
     xAzResult = az_iot_adu_client_get_service_properties_response(
         &pxAzureIoTADUClient->_internal.xADUClient,
         ( int32_t ) ulPropertyVersion,
-        200,
+        xRequestDecision == eAzureIoTADURequestDecisionAccept ? 200 : 406,
```

Comment:
What do you think about defining this values (200, 406) as macros at the top, with some self-explanatory names?

Suggestion:

```c
#define AZURE_IOT_ADU_REQUEST_ACCEPTED 200
#define AZURE_IOT_ADU_REQUEST_REJECTED 406
```

Reply:
yes yes definitely